src/commands.ts

```diff
@@ -428,6 +431,50 @@ function openInExternalEditor(packageNode: PackageNode) {
     }
 }
 
+/**
+ * Choose DEVELOPER_DIR
+ * @param workspaceContext
+ */
+async function selectXcodeDeveloperDir() {
+    const defaultXcode = await SwiftToolchain.getXcodeDeveloperDir();
+    const selectedXcode = configuration.swiftEnvironmentVariables.DEVELOPER_DIR;
+    const xcodes = await SwiftToolchain.getXcodeInstalls();
+    await withQuickPick(
+        selectedXcode ?? defaultXcode,
+        xcodes.map(xcode => {
+            const developerDir = `${xcode}/Contents/Developer`;
+            return {
+                label: developerDir === defaultXcode ? `${xcode} (default)` : xcode,
+                folder: developerDir === defaultXcode ? undefined : developerDir,
+            };
+        }),
+        async selected => {
+            let swiftEnv = configuration.swiftEnvironmentVariables;
+            const previousDeveloperDir = swiftEnv.DEVELOPER_DIR ?? defaultXcode;
+            if (selected.folder) {
+                swiftEnv.DEVELOPER_DIR = selected.folder;
+            } else if (swiftEnv.DEVELOPER_DIR) {
+                // if DEVELOPER_DIR was set and the new folder is the default then
+                // delete variable
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                const { DEVELOPER_DIR, ...rest } = swiftEnv;
+                swiftEnv = rest;
+            }
+            configuration.swiftEnvironmentVariables = swiftEnv;
+            // if SDK is inside previous DEVELOPER_DIR then move to new DEVELOPER_DIR
+            if (
+                configuration.sdk.length > 0 &&
```

Comment:
Here is convenient enough 

but I'm wondering if it's possible for the setting of swift.SDK could be like either way
1.  swift.SDK: "SDKs/MacOSX.sdk"  (`DEVELOPER_DIR` is the automatic prefix for it)
2. swift.SDK: "`$DEVELOPER_DIR`/SDKs/MacOSX.sdk".  (environment variable can be used in arguments too)

a bit out of scope here. you may find some ideas here

Reply:
DEVELOPER_DIR doesn't exist on Linux or Windows. Also the SDK can exist outside the DEVELOPER_DIR if you are cross compiling for another platform.

Reply:
Thanks for your answer